test/e2e/operator.go

```diff
@@ -433,10 +434,20 @@ var _ = Describe("ARO Operator - ImageConfig Reconciler", func() {
 	var requiredRegistries []string
 	var imageconfig *configv1.Image
 
-	getLocationRegistries := func(acrDomain, acrDnsSuffix, location string) []string {
-		regionalAcr := fmt.Sprintf("arosvc.%s.data.%s", location, acrDnsSuffix)
-		requiredRegistries := []string{acrDomain, regionalAcr}
-		return requiredRegistries
+	// Reimplementation of function from image config controller
+	getCloudAwareRegistries := func(instance *arov1alpha1.Cluster) ([]string, error) {
+		var requiredRegistries []string
+		switch instance.Spec.AZEnvironment {
+		case azureclient.PublicCloud.Environment.Name:
+			requiredRegistries = []string{instance.Spec.ACRDomain, "arosvc." + instance.Spec.Location + ".data." + azure.PublicCloud.ContainerRegistryDNSSuffix}
+
+		case azureclient.USGovernmentCloud.Environment.Name:
+			requiredRegistries = []string{instance.Spec.ACRDomain, "arosvc." + instance.Spec.Location + ".data." + azure.USGovernmentCloud.ContainerRegistryDNSSuffix}
+
+		default:
+			return nil, fmt.Errorf("cloud environment %s is not supported", instance.Spec.AZEnvironment)
```

Comment:
I agree we should fail e2e if environment is unsupported (or if environment is blank/deleted), but not sure if we should fail it during this test specifically.

Reply:
That's a good point but I think I actually like it the way it is now. There's already a check in the `BeforeEach()` that will skip the tests if the feature flag is unset, having an error here will tell us if the flag is enabled on an unsupported cloud environment for whatever reason.